projects/storefrontlib/src/layout/a11y/keyboard-focus/on-navigate/on-navigate-focus.service.ts

```diff
@@ -13,35 +15,90 @@ import { KeyboardFocusConfig } from '../config';
 })
 export class OnNavigateFocusService {
   protected resetFocusOnNavigate: Subscription;
+  protected resetViewOnNavigate: Subscription;
 
   constructor(
     protected config: KeyboardFocusConfig,
     protected router: Router,
-    @Inject(DOCUMENT) protected document: any
+    @Inject(DOCUMENT) protected document: any,
+    protected breakpointService: BreakpointService
   ) {}
 
   /**
    * Reads configuration and enables features based on flags set.
    */
   initializeWithConfig(): void {
     if (this.config?.keyboardFocus?.enableResetFocusOnNavigate) {
-      this.setResetFocusOnNavigate();
+      this.setResetFocusOnNavigate(
+        this.config.keyboardFocus.enableResetFocusOnNavigate
+      );
+    }
+
+    if (this.config?.keyboardFocus?.enableResetViewOnNavigate) {
+      this.setResetViewOnNavigate(
+        this.config.keyboardFocus.enableResetViewOnNavigate
+      );
     }
   }
 
   /**
    * Resets focus back to root element `<cx-storefront>` in the DOM tree when a navigation is started.
-   * @param enable (default: true): Enable or disable this feature.
+   * @param enable Enable or disable this feature. Set this to a BREAKPOINT value string to enable only up to the specified screen width.
    */
-  setResetFocusOnNavigate(enable = true): void {
+  setResetFocusOnNavigate(enable: boolean | BREAKPOINT): void {
     this.resetFocusOnNavigate?.unsubscribe();
 
     if (enable) {
       this.resetFocusOnNavigate = this.router.events
         .pipe(filter((event) => event instanceof NavigationStart))
         .subscribe(() => {
-          this.document.getElementsByTagName('cx-storefront')[0].focus();
+          if (typeof enable === 'string') {
+            this.breakpointService
+              .isDown(<BREAKPOINT>enable)
+              .pipe(take(1))
+              .subscribe((isDown: boolean) => {
+                if (isDown) {
+                  this.getStorefrontElement().focus();
+                }
+              });
+          } else if (typeof enable === 'boolean') {
+            this.getStorefrontElement().focus();
+          }
+        });
+    }
+  }
+
+  /**
+   * Resets view back to root element `<cx-storefront>` in the DOM tree when a navigation is started.
+   * @param enable Enable or disable this feature. Set this to a BREAKPOINT value string to enable only up to the specified screen width.
```

Comment:
I'm not sure how big limitation it is, but in this approach if customer would like to setup this behaviour for tablet and mobile screen sizes only (lg, md, sm) he is not able to do that.

Worth to consider might be using array of breakpoints as next alternative?

Reply:
In this instance, the user would use the breakpoint `lg` and this works for all breakpoints from `lg` downwards (ie. `lg` `md` `sm` `xs`).